presto-verifier/src/main/java/com/facebook/presto/verifier/rewrite/FunctionCallRewriter.java

```diff
@@ -0,0 +1,136 @@
+import com.facebook.presto.sql.tree.QualifiedName;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+import static com.google.common.collect.ImmutableList.toImmutableList;
+import static java.util.Objects.requireNonNull;
+
+public class FunctionCallRewriter
+        extends DefaultTreeRewriter<Void>
+{
+    public static class FunctionCallSubstitute
+    {
+        private final QualifiedName name;
+        private final List<Integer> originalArgumentIndices;
+
+        public FunctionCallSubstitute(QualifiedName name, List<Integer> originalArgumentIndices)
+        {
+            this.name = requireNonNull(name, "name is null");
+            this.originalArgumentIndices = ImmutableList.copyOf(requireNonNull(originalArgumentIndices, "originalArgumentIndices is null"));
+        }
+
+        @Override
+        public boolean equals(Object o)
+        {
+            if (this == o) {
+                return true;
+            }
+            if (o == null || getClass() != o.getClass()) {
+                return false;
+            }
+            FunctionCallSubstitute that = (FunctionCallSubstitute) o;
+            return name.equals(that.name) && Objects.equals(originalArgumentIndices, that.originalArgumentIndices);
+        }
+
+        @Override
+        public int hashCode()
+        {
+            return Objects.hash(name, originalArgumentIndices);
+        }
+    }
+
+    private static final Pattern FUNCTION_CALL_SUBSTITUTION_PATTERN = Pattern.compile("/(\\w+)\\(([\\w|,]+)\\)/(\\w+)\\(([\\w|,]+)\\)/");
+
+    private final Map<QualifiedName, FunctionCallSubstitute> functionCallSubstituteMap;
+
+    private FunctionCallRewriter(Map<QualifiedName, FunctionCallSubstitute> functionCallSubstituteMap)
+    {
+        this.functionCallSubstituteMap = requireNonNull(functionCallSubstituteMap, "functionCallSubstituteMap is null.");
+    }
+
+    public static FunctionCallRewriter getInstance(String functionCallSubstitutes)
+    {
+        Map<QualifiedName, FunctionCallSubstitute> functionCallSubstituteMap = constructFunctionCallSubstituteMap(functionCallSubstitutes);
+        if (functionCallSubstituteMap == null || functionCallSubstituteMap.isEmpty()) {
+            return null;
+        }
+        return new FunctionCallRewriter(functionCallSubstituteMap);
+    }
+
+    public static Map<QualifiedName, FunctionCallSubstitute> constructFunctionCallSubstituteMap(String functionCallSubstitutes)
+    {
+        ImmutableMap.Builder<QualifiedName, FunctionCallSubstitute> map = ImmutableMap.builder();
+        if (functionCallSubstitutes == null) {
+            return map.build();
+        }
+
+        Matcher matcher = FUNCTION_CALL_SUBSTITUTION_PATTERN.matcher(functionCallSubstitutes);
+
+        while (matcher.find()) {
```

Comment:
What happens if someone specified incorrect value that cannot be parsed / matched? Do we silently ignore the config? Can we fail instead?

Reply:
Yeah, right now it fails silently.

I've considered explicit failure. Will it be okay if we fail the presto verifier start, since it's a config for all test cases and failing individual test cases when they start will be inefficient?

Reply:
Yes, failing as early as possible is best.